### ISSUE TITLE
memberLookup doesn't support member of a function

### ISSUE THREAD

**Issue by breeswish** [mozilla/nunjucks#884]
For example, a static member of a class:

```js
var Foo = function () { ... };
Foo.var_x = 'bar';
```

`memberLookup` treats `Foo` as a function and cannot lookup its members (`var_x`).

**Comment by jbmoelker** [comment]
I'm unfamiliar with `memberLookup`. I understand it's `jinjaCompat` feature. Correct? Do you know how Jinja handles this? Could you explain where you need it for? It's hard for me to deduce the need for `memberLookup` as it also appears to lack test coverage and documentation.

**Comment by breeswish** [comment]
@jbmoelker in fact `memberLookup` will be used in template parsing.

Let's say, we have a class called `Foo` and it has static members called `var_x`:

```js
var Foo = function () { ... };
Foo.var_x = 'bar';
```

Now, I would like to access `Foo` in the template so I will write something like this:

```js
njenv.addGlobal('Foo', Foo);
```

And access it like this:

```jinja
{{ Foo.var_x }}
```

However the code above will fail because `memberLookup` treat `Foo` as a function and `{{ Foo.var_x }}` becomes `Foo().var_x`.

**Comment by fdintino** [comment]
That seems to me like it might be a bug. That said, what should work would be something like:

```javascript
var Foo = function() {};
Foo.prototype.var_x = 'bar';

env.addGlobal('foo', new Foo());
```

and

```jinja2
{{ foo.var_x }}
```

**Comment by breeswish** [comment]
Yeah. This is a workaround for this bug. In my case, my application has some services (classes) which will be initialized (`new xxx()`) by our dependency injection framework. Due to this bug, I cannot directly access static members (usually constants) defined by services. I'm currently using a helper function to handle this issue which will correctly resolve static members. Not elegant of course. Hope to see this issued fixed.

**Comment by undefined-moe** [comment]
This works for me:

src/runtime.js#L246

```js
function memberLookup(obj, val) {
  if ((obj || {})._original) obj = obj._original;
  if (obj === undefined || obj === null) return undefined;
  if (typeof obj[val] === 'function') {
    const fn = function () {
      for (var _len2 = arguments.length, args = new Array(_len2), _key2 = 0; _key2 < _len2; _key2++) {
        args[_key2] = arguments[_key2];
      }
      return obj[val].apply(obj, args);
    };
    fn._original = obj[val];
    return fn;
  }
  return obj[val];
}
```
